examples/ecs/hierarchy.rs

```diff
@@ -0,0 +1,112 @@
+                    texture: Some(texture),
+                }),
+                ..Default::default()
+            });
+        })
+        // Store parent entity for next sections
+        .current_entity()
+        .unwrap();
+
+    // Another way to create a hierarchy is to add a Parent component to an entity,
+    // which would be added automatically to parents with other methods.
+    // Similarly, adding a Parent component will automatically add a Children component to the parent.
+    commands
+        .spawn(SpriteComponents {
+            transform: Transform::from_translation(Vec3::new(-250.0, 0.0, 0.0)).with_scale(0.75),
+            material: materials.add(ColorMaterial {
+                color: Color::RED,
+                texture: Some(texture),
+            }),
+            ..Default::default()
+        })
+        // Using the entity from the previous section as the parent:
+        .with(Parent(parent));
+
+    // Another way is to use the push_children function to add children after the parent
+    // entity has already been spawned.
+    let child = commands
+        .spawn(SpriteComponents {
+            transform: Transform::from_translation(Vec3::new(0.0, 250.0, 0.0)).with_scale(0.75),
+            material: materials.add(ColorMaterial {
+                color: Color::GREEN,
+                texture: Some(texture),
+            }),
+            ..Default::default()
+        })
+        .current_entity()
+        .unwrap();
+
+    // Pushing takes a slice of children to add:
+    commands.push_children(parent, &[child]);
+}
+
+// A simple system to rotate the root entity, and rotate all its children separately
+fn rotate(
+    mut commands: Commands,
+    time: Res<Time>,
+    mut parents_query: Query<(&mut Transform, &mut Children, &Sprite)>,
+    children_query: Query<(&mut Transform, &Sprite)>,
+) {
+    let angle = std::f32::consts::PI / 2.0;
+    for (mut transform, mut children, _) in &mut parents_query.iter() {
+        transform.rotate(Quat::from_rotation_z(-angle * time.delta_seconds));
+
+        // To iterate through the entities children, just treat the Children component as a Vec
+        // Alternatively, you could query entities that have a Parent component
+        for child in children.iter() {
+            let mut transform = children_query.get_mut::<Transform>(*child).unwrap();
+            transform.rotate(Quat::from_rotation_z(angle * 2.0 * time.delta_seconds));
+        }
+
+        // To demonstrate removing children, we'll start to remove the children after a couple of seconds
+        if time.seconds_since_startup >= 2.0 && children.len() == 3 {
+            // Using .despawn() on an entity does not remove it from its parent's list of children!
+            // It must be done manually if using .despawn()
+            let child = children.pop().unwrap();
+            commands.despawn(child);
+        }
+
+        if time.seconds_since_startup >= 4.0 && children.len() == 2 {
```

Comment:
To better illustrate the "despawn recursive" behavior, it might be better to despawn the "parent" entity

Reply:
That's a good point! Although I do like the idea of showing that despawn_recursive also takes care of cleaning up the parent's children. Maybe keep this one in and add another step where it despawns the parent afterwards? As well as some clearer comments to go with it.

Reply:
The issue is that we aren't directly illustrating `despawn_recursive` if it doesn't despawn multiple entities. If we leave the current line as-is, it only despawns one entity.  If after the fact we despawn the parent recursively, that also only despawns one entity because there will be no children left.

I think this has my preference:

```rust
        // To demonstrate removing children, we'll start to remove the children after a couple of seconds
        if time.seconds_since_startup >= 2.0 && children.len() == 3 {
            // Using .despawn() on an entity does not remove it from its parent's list of children!
            // It must be done manually if using .despawn()
            // NOTE: this is a bug. eventually Bevy will update the children list automatically
            let child = children.pop().unwrap();
            commands.despawn(child);
        }

        if time.seconds_since_startup >= 4.0 {
            // Alternatively, you can use .despawn_recursive()
            // This will remove the entity from its parent's list of children, as well as despawn
            // any children the entity has.
            commands.despawn_recursive(parent);
        }
```

I added a "note" that explains that the "children list maintenance" will eventually be fixed